carp.common/src/commonMain/kotlin/dk/cachet/carp/common/RecurrenceRule.kt

```diff
@@ -43,13 +45,62 @@ data class RecurrenceRule(
         fun weekly( interval: Int = 1, end: End = End.Never ) = RecurrenceRule( Frequency.WEEKLY, interval, end )
         fun monthly( interval: Int = 1, end: End = End.Never ) = RecurrenceRule( Frequency.MONTHLY, interval, end )
         fun yearly( interval: Int = 1, end: End = End.Never ) = RecurrenceRule( Frequency.YEARLY, interval, end )
-    }
 
-    init
-    {
-        require( interval >= 1 ) { "Interval needs to be 1 or more." }
+        /**
+         * Initialize a [RecurrenceRule] based on a [rrule] string.
+         */
+        fun fromString( rrule: String ): RecurrenceRule
+        {
+            require( RecurrenceRuleRegex.matches( rrule ) ) { "Invalid or unsupported RecurrenceRule string representation." }
```

Comment:
Just a thought. We could define validators as external functions `f(input: String): Boolean` - leaving you with a single line require in the init block and a reusable function that can be used elsewhere like UI input validation.

Reply:
Isn't this an 'external' function? This is defined in a companion object. The only difference would be it would return true/false instead of `RecurrenceRule`. So, if there is ever a use case in which you expect erroneously formatted RRule's this would make sense.

I don't see that right now.